diff --git a/src/ui/ui_passwords.c b/src/ui/ui_passwords.c
index 8ca9918..d0f8092 100644
--- a/src/ui/ui_passwords.c
+++ b/src/ui/ui_passwords.c
@@ -178,10 +178,10 @@ void confirm_all_passwords_deletion(void) {
  *
  */
 static void password_callback(const int token, const uint8_t index, int page) {
     UNUSED(token);
-    if (selector_callback) {
-        selector_callback((page * nbPasswordsPerPage) + index);
-    }
+    UNUSED(page);
+    if (selector_callback) {
+        selector_callback(index);
+    }
 }
 
 /**
@@ -224,9 +224,6 @@ void display_password_list(void) {
  #ifdef HAVE_PIEZO_SOUND
     contentsList.content.choicesList.tuneId = TUNE_TAP_CASUAL;
  #endif
     contentsList.contentActionCallback = &password_callback;
-    nbPasswordsPerPage =
-        nbgl_useCaseGetNbChoicesInPage(nbPasswords, &contentsList.content.choicesList, 0, false);
-
     nbgl_useCaseGenericConfiguration("Passwords list", 0, &genericContent, display_choice_page);
 }
